src/ng/compile.js

```diff
@@ -3836,8 +3836,13 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
           return {
             pre: function ngPropPreLinkFn(scope, $element) {
               function applyPropValue() {
-                var propValue = ngPropGetter(scope);
-                $element.prop(propName, sanitizer(propValue));
+                var propValue = sanitizer(ngPropGetter(scope));
+
+                if (propValue !== undefined) {
+                  $element.prop(propName, propValue);
+                } else {
+                  $element.removeProp(propName);
```

Comment:
Does not exist in jqLite :grin:

Reply:
Should we add it?

Reply:
Actually looking at what it does in jQuery, I don't think this is what we need. I'd rather avoid complication with built-in vs custom properties etc and just set the prop to `undefined`(i.e. not `delete` it). WDYT?

Reply:
Yeah I think just set undefined directly on the element, but still use `.prop` to set non-undefined values.

Reply:
Why can't you use `prop` in all cases?

Reply:
Passing `undefined` makes it a getter instead of setter, that's the bug.

Reply:
I've updated this to use `.removeProp` if available, otherwise do `delete $element[0][propName]`. This way if jQuery is being used then the proper hooks will run, just like they do on `.prop`.

I think it should be `delete` instead of assigning `undefined` because that's what jQuery does...